h2/src/main/org/h2/mvstore/tx/TransactionMap.java

```diff
@@ -650,9 +650,10 @@ public K lowerKey(K key) {
                 return new UncommittedIterator<>(this, from, to, false, forEntries);
             case REPEATABLE_READ:
             case SERIALIZABLE:
-                return transaction.hasChanges() ?
-                        new RepeatableIterator<K, X>(this, from, to, forEntries) :
-                        new CommittedIterator<K, X>(this, from, to, forEntries);
+                if (transaction.hasChanges()) {
+                    new RepeatableIterator<K, X>(this, from, to, forEntries);
```

Comment:
Missing `return` before `new`.

Reply:
And you can replace type parameters with diamond `<>`.

Unfortunately, Java 7 requires type parameters in different strange places.